### ISSUE TITLE
GPU time left is wildly inaccurate for projects with more than one GPU app

### ISSUE THREAD

**Issue by hucker75** [BOINC/boinc#5011]
**Describe the bug**
For projects such as Primegrid with many GPU apps, if Boinc is used to seeing one of them, then gets a different one, the "time left" estimate is wildly inaccurate.

For example, I currently have a genefer_extreme task which has reached about halfway in 2 days.  So sensibly, we'd say the other half would take two days aswell.  But it has an estimate of 36 days.  This completely screws up scheduling for other projects on that GPU, as it believes the Primegrid task will not reach the deadline, and enters panic mode.

**Steps To Reproduce**
1. Run Primegrid on small tasks on GPU, such as genefer 15 (the smallest GPU task).
2. Switch Primegrid to do genefer_extreme (the largest GPU task).
3. Let it run for a bit and observe the crazy estimate of remaining time.

**System Information**
 - OS: Windows 11
 - BOINC Version: 7.20.2

**Additional context**
Perhaps Boinc could concentrate less on assuming the task could be non-linear in progress, and concentrate more on how long this one has taken so far?  Or perhaps record how long each app takes?  It would appear currently Boinc is not allowing for multiple GPU apps from one project.

**Comment by RichardHaselgrove** [comment]
If progress is linear, use `<fraction_done_exact/>` in an [app_config.xml](https://boinc.berkeley.edu/wiki/Client_configuration#Project-level_configuration) file.

And tell the project to pay attention to <rsc_fpops_est> when setting up the task template file - it's a project problem, not BOINC's.

**Comment by hucker75** [comment]
I thought Boinc didn't support multiple apps under GPU for time estimates?  Has this changed?

But I'll try fraction done exact, sounds useful.

**Comment by RichardHaselgrove** [comment]
It was changed in [October 2010](https://boinc.berkeley.edu/trac/wiki/ClientSchedOctTen).

Any project which shows - and uses - an "Average processing rate" in a host's Application details (i.e. most of them, but **excluding** Einstein) is equipped to do that.

Note that boundary conditions apply: a new application or app_version won't show realistic estimates until the host has returned 11 "successful" tasks. Successful requires a normal exit plus validation.

**Comment by hucker75** [comment]
Surely until it has 11 successful tasks, it should estimate according to the fraction done system?  Whatever it uses by default is wildly wrong.  And 11 tasks is far too long to wait for week-long genefer_extreme tasks.

**Comment by AenBleidd** [comment]
I don't think we have an issue here but it worth to keep it open, so I convert this to discussion
